lib/actions/user.js

```diff
@@ -93,10 +103,57 @@ export function createOrUpdateUser (userData) {
 
       // TODO: improve the UI feedback messages for this.
       if (result.status === 'success' && result.data) {
+        alert('Your preferences have been saved.')
+
         // Update application state with the user entry as saved
         // (as returned) by the middleware.
         const userData = result.data
         dispatch(setCurrentUser({ accessToken, user: userData }))
+      } else {
+        alert(`An error was encountered:\n${JSON.stringify(result)}`)
+      }
+    }
+  }
+}
+
+/**
+ * Updates a logged-in user's monitored trip,
+ * then, if that was successful, refreshes the redux monitoredTrips
+ * with the updated trip.
+ */
+export function createOrUpdateUserMonitoredTrip (tripData, isNew) {
+  return async function (dispatch, getState) {
+    const { otp, user } = getState()
+    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
+
+    if (otpMiddleware) {
+      const { accessToken, loggedInUserMonitoredTrips: trips } = user
+
+      let result
+      if (isNew) {
+        result = await addTrip(otpMiddleware, accessToken, tripData)
+      } else {
+        result = await updateTrip(otpMiddleware, accessToken, tripData)
+      }
+
+      // TODO: improve the UI feedback messages for this.
+      if (result.status === 'success' && result.data) {
+        const newTrip = result.data
+
+        // For updating the monitoredTrips state,
+        // make a clone of the trips array,
+        // find the index of the trip being updated (using trip.id),
+        // and assign the trip returned from the middleware at that index
+        // (or append to array if index was not found).
+        const newTrips = trips ? clone(trips) : []
```

Comment:
Let's just dispatch `fetchTrips` or whatever the equivalent method is.

Reply:
Addressed in https://github.com/opentripplanner/otp-react-redux/pull/174/commits/b0395b10a66ab3639f243d25250007599aaf70dd.